Joystream-Role-Bot/src/controls/control.ts

```diff
@@ -132,32 +154,65 @@ export const setMemberRole = async (client: Client): Promise<void> => {
     //   console.error("Error removing member roles:", error);
     // }
 
-    await discordMember.roles.add(RoleAddress.membershipLinked);
 
-    const roleUpdatePromises = qnMember.roles.map(async (memberRole) => {
-      const memberRoleGroupId = memberRole.groupId;
-      const mappedRoles = roleMap[memberRoleGroupId];
-      if (!mappedRoles) return;
+    const getRoles = discordMember.roles.cache.map((role: any) => role.id)
 
-      const [leadRoleId, workerRoleId] = mappedRoles;
-      const roleId = memberRole.isLead ? leadRoleId : workerRoleId;
-      const role = await guild.roles.fetch(roleId);
+    if (!getRoles.find((id: string) => id === RoleAddress.membershipLinked)) {
+      await discordMember.roles.add(linkRole);
+    }
 
-      if (!role) {
-        console.log(`<@&${roleId}> Role not found`);
-        return;
-      }
+    let nowWorking: GetWorkingState[] = [];
+
+    const roleUpdatePromises = qnMember.roles.map(async (memberRole: any) => {
+
+      const index = nowWorking.findIndex(data => data.workingGroups === memberRole.groupId)
```

Comment:
This logic doesn't resolve some of the issues reported. You are going through all the roles of the user ever and then overwriting with the most recent one. For example, here are my roles in DWG:
```
          {
            "status": {
              "__typename": "WorkerStatusTerminated"
            },
            "groupId": "distributionWorkingGroup",
            "isLead": false
          },
          {
            "status": {
              "__typename": "WorkerStatusActive"
            },
            "groupId": "distributionWorkingGroup",
            "isLead": false
          },
          {
            "status": {
              "__typename": "WorkerStatusTerminated"
            },
            "groupId": "distributionWorkingGroup",
            "isLead": false
          },
          {
            "status": {
              "__typename": "WorkerStatusActive"
            },
            "groupId": "distributionWorkingGroup",
            "isLead": false
          },
          {
            "status": {
              "__typename": "WorkerStatusTerminated"
            },
            "groupId": "distributionWorkingGroup",
            "isLead": false
          },
```
As you can see, last role is terminated, but I still have 2 active roles in DWG. You cannot only consider the last role.